client/app/bundles/course/assessment/submission/components/answers/Programming/index.jsx

```diff
@@ -1,59 +1,69 @@
 import 'ace-builds/src-noconflict/mode-python';
 import 'ace-builds/src-noconflict/theme-github';
-
-import { Component } from 'react';
 import PropTypes from 'prop-types';
-import { FieldArray } from 'redux-form';
+import { useFieldArray, useFormContext, useWatch } from 'react-hook-form';
 import ProgrammingFile from './ProgrammingFile';
 import ProgrammingImportEditor from '../../../containers/ProgrammingImportEditor';
 import TestCaseView from '../../../containers/TestCaseView';
 import { parseLanguages } from '../../../utils';
 import { questionShape } from '../../../propTypes';
 
-class Programming extends Component {
-  static renderProgrammingFiles({ fields, readOnly, answerId, language }) {
-    return (
-      <>
-        {fields.map((fieldName, index) => {
-          const file = fields.get(index);
-          return (
-            <ProgrammingFile
-              key={file.id}
-              {...{ file, fieldName, readOnly, answerId, language }}
-            />
-          );
-        })}
-      </>
-    );
-  }
+const renderProgrammingFiles = ({ readOnly, answerId, language }) => {
```

Comment:
If this is more or less a component in itself, can rename it as a component and use it like a component?

Otherwise, if this is designed to be a function, don't use hooks inside: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

Reply:
Noted and thanks for pointing this out!